lib/parse.js

```diff
@@ -919,13 +925,21 @@ function parse($TEXT, options) {
         if (!is("punc", ";")) {
             init = is("keyword", "var")
                 ? (next(), var_(true))
-                : expression(true, true);
+                : is("keyword", "let")
```

Comment:
good question, 
and I test `var let={a:1};`, `let let={a:1}`, `let var={a:1}`, 
only the first pass in io.js 2.0. 

it seems that the `let` is not treated as an reserved keyword ,but in LetExpression is cannot contain `let`.

maybe it is because ES6 need back compatible with ES5.